### PR TITLE
Fix image overlay rescale

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by huffman** [comment]
The move to using TextureCache for ImageOverlay images broke ImageOverlay when an image was over the texture size limit (1024x1024) and used subimages.  This update fixes that issue by taking the original image size into account to scale the subimage proportionately.

File: libraries/render-utils/src/TextureCache.cpp [hunk 1]
```diff
@@ -531,9 +535,11 @@ void NetworkTexture::loadContent(const QByteArray& content) {
     QThreadPool::globalInstance()->start(new ImageReader(_self, NULL, _url, content));
 }
 
-void NetworkTexture::setImage(const QImage& image, bool translucent, const QColor& averageColor) {
+void NetworkTexture::setImage(const QImage& image, bool translucent, const QColor& averageColor, int originalWidth, int originalHeight) {
```
Comment (thoys) [inline libraries/render-utils/src/TextureCache.cpp]: http://docs.highfidelity.io/v1.0/docs/coding-standard 
2.1.4. File content must be kept within 128 columns.

File: libraries/render-utils/src/TextureCache.h [hunk 2]
```diff
@@ -158,7 +160,7 @@ class NetworkTexture : public Resource, public Texture {
     virtual void downloadFinished(QNetworkReply* reply);
           
     Q_INVOKABLE void loadContent(const QByteArray& content);
-    Q_INVOKABLE void setImage(const QImage& image, bool translucent, const QColor& averageColor);
+    Q_INVOKABLE void setImage(const QImage& image, bool translucent, const QColor& averageColor, int originalWidth, int originalHeight);
```
Comment (thoys) [inline libraries/render-utils/src/TextureCache.h]: http://docs.highfidelity.io/v1.0/docs/coding-standard 
2.1.4. File content must be kept within 128 columns.

File: libraries/render-utils/src/TextureCache.cpp (hunk 1 above)
Comment (huffman) [inline libraries/render-utils/src/TextureCache.cpp]: Thanks, fixed

File: libraries/render-utils/src/TextureCache.h (hunk 2 above)
Comment (huffman) [inline libraries/render-utils/src/TextureCache.h]: Fixed

**Comment by OmegaHeron** [comment]
If this is level where windows 1821 springs from... take a look at it on windows.  I'm seeing no entities displayed that have textures assigned.  My trees (very over complex geometries) rez but have only colors assigned, no textures.  @huffman

**Pull request merged**